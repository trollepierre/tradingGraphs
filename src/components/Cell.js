import React from 'react';

const Cell = props => {
  function truncate(number) {
    return Math.round(100 * number) / 100;
  }

  function display(value, columnName) {
    return truncate(value.stocks[columnName]);
  }

  const paddedTd = {
    padding: '5px',
  };

  return (
    <td style={paddedTd}>
      {display(props.value, props.type)}
    </td>
  );
};

export default Cell;