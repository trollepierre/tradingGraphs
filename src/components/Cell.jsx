import React from 'react';
import EditableCell from './EditableCell.jsx';
import RenderingCell from './RenderingCell.jsx';
import { connect } from 'react-redux';
import { updateTradingValue } from '../store/actions/updateTradingValueAction';

class Cell extends React.Component {
  constructor(props) {
    super(props);
    this.editModeChange = this.editModeChange.bind(this);
    this.updateCellValue = this.updateCellValue.bind(this);
    this.state = { isEditMode: props.isEditMode }
  }

  editModeChange() {
    console.log('in edit mode');
    this.setState({ isEditMode: true });
  }

  updateCellValue(newValue) {
    console.log(newValue);
    const { dispatch, value, type } = this.props;
    dispatch(updateTradingValue(newValue, value.index, type));
  }

  render() {
    return this.state.isEditMode ?
      <EditableCell onNewValue={this.updateCellValue} />
      :
      <RenderingCell
        onEditMode={this.editModeChange}
        value={this.props.value}
        type={this.props.type}
      />;
  }
}

export default connect()(Cell);
