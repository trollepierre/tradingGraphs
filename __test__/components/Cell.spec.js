import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import Cell from '../../src/components/Cell'

const value = { "timestamp": 1522088675779, "index": 1171, "stocks": { "NASDAQ": 5.4068, "CAC40": 39.45096 } }

// Snapshot for Cell React Component
xdescribe('>>>CELL --- Snapshot', () => {
  it('+++capturing Snapshot of Cell', () => {
    const renderedValue = renderer.create(<Cell type='CAC40' value={value}/>).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });

});

xdescribe('>>>CELL --- Shallow Render REACT COMPONENTS', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Cell type='CAC40' value={value}/>)
  })

  it('+++ render the DUMB component', () => {
    // expect(wrapper.length).toEqual(1)
  });

  it('+++ contains td CAC40', () => {
    // expect(wrapper.find('td').at(0).nodes[0].key).toBe('CAC40-1171');
    // expect(wrapper.find('td').at(0).nodes[0].props.children).toBe(39.45);
  });
});
