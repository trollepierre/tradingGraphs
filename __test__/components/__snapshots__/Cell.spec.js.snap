exports[`>>>CELL --- Snapshot +++capturing Snapshot of Cell 1`] = `
<td
  style={
    Object {
      "padding": "5px",
    }
  }>
  39.45
</td>
`;
